### PR TITLE
Convert MavenFilteringHack, Groovy to Java

### PR DESCRIPTION
**Pull request by edh-oss**
Port to Java, related to issue #34459

### CODE REVIEW

**Comment by breskeby** [comment]
@elasticsearchmachine test this please

File: buildSrc/src/main/java/org/elasticsearch/gradle/MavenFilteringHack.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+public class MavenFilteringHack {
+    /**
+     * Adds a filter to the given copy spec that will substitute maven variables.
+     * 
+     */
+    static void filter(CopySpec copySpec, Map<String,String> substitutions) {
+        Map<String,String> mavenSubstitutions = new LinkedHashMap<>();
+        Map<String, Object> argMap = new LinkedHashMap<>();
+
+        substitutions.forEach((k,v) -> mavenSubstitutions.put("{" + k, v));
```
Comment (breskeby) [inline buildSrc/src/main/java/org/elasticsearch/gradle/MavenFilteringHack.java:35]: v needs to be explicitly converted to a String here

**Review by breskeby — CHANGES_REQUESTED** [review]
Hello,

thanks for your contribution. There's a minor issue with your changes that breaks the build when passing non String values to the `MavenFilteringHack`.

If you are able to fix this we can merge this PR

File: buildSrc/src/main/java/org/elasticsearch/gradle/MavenFilteringHack.java (hunk 1 above)
Comment (edh-oss) [reply buildSrc/src/main/java/org/elasticsearch/gradle/MavenFilteringHack.java]: I've updated the code to do what the groovy code was doing, explicitly calling toString() on the incoming Map keys and values.  I've also updated the type arguments in the method signature, to indicate that we don't know what types of objects will be contained in the Map argument.

**Comment by breskeby** [comment]
@elasticmachine test this please

**Comment by breskeby** [comment]
@elasticmachine test this please

**Comment by breskeby** [comment]
@edh-oss I've merged your PR. Thanks again for your contribution!

**Pull request merged**